okhttp/src/main/java/okhttp3/EventListener.java

```diff
@@ -54,11 +54,27 @@ public void dnsEnd(Call call, String domainName, @Nullable List<InetAddress> ine
   public void connectStart(Call call, InetAddress address, int port) {
   }
 
+  /**
+   * Invoked just prior to initiating a TLS connection.
+   *
+   * <p>If the {@link Call#request()} does not use TLS, this method will not be invoked.
+   */
   public void secureConnectStart(Call call) {
   }
 
-  public void secureConnectEnd(Call call, Handshake handshake,
-      Throwable throwable) {
+  /**
+   * Invoked immediately after a TLS connection was attempted.
+   *
+   * <p>If the {@link Call#request()} does not use TLS, this method will not be invoked.
```

Comment:
Without this event you don't know the handshake details when reusing a connection, and you don't know if the handshake changes when your new connection is closed as a dupe